specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-07-01/virtualMachine.json

```diff
@@ -1891,6 +1891,72 @@
           "final-state-via": "location"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/attachDetachDataDisks": {
+      "post": {
+        "tags": [
+          "VirtualMachines"
+        ],
+        "operationId": "VirtualMachines_AttachDetachDataDisks",
+        "description": "Attach and detach data disks to/from the virtual machine.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "vmName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the virtual machine."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./computeRPCommon.json#/definitions/AttachDetachDataDisksRequest"
+            },
+            "description": "Parameters supplied to the attach and detach data disks operation on the virtual machine."
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "202": {
```

Comment:
Async flow is not required if there is no change between current state and goal state.

- Should we have a common success status as 200 (as is the case for exiting Create/Update API (https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update?tabs=HTTP) 
or
- should we have 2 separate success status i.e 200 if there is no change and 202 if there is change and async operation is triggered.

Reply:
> My suggestion is to have two seperate responses. Reason is 202 represents "Request-Accepted" and on background the request will be processed. "200" represents the state of the entity has not been modified and no background processing would take place . In case of 200, I guess the header will also not have any operation status. Correct me, if I am missing something